pkg/generate/app/cmd/newapp.go

```diff
@@ -511,7 +518,11 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						}
 					}
 				}
-				if pipeline, err = app.NewBuildPipeline(ref.Input().String(), input, c.OutputDocker, strategy, source); err != nil {
+				buildEnvs := app.Environment{}
+				if c.BuildEnvironment != nil {
+					buildEnvs = environment
+				}
```

Comment:
This doesn't look good. You only check c.BuildEnvironment being non-null, then set env to the environment argument... I think it would be better to pass a separate buildEnvironment parameter to this method that is derived from c.BuildEnvironment.